### PR TITLE
Fixed Prevent Unwanted Spaces in Bounty Fields During Creation

### PR DESCRIPTION
**Pull request by aliraza556**
### Problem:
The platform currently allows the creation of bounties with titles, descriptions, and acceptance criteria containing only spaces. This behavior results in bounties that may seem empty or improperly formatted, leading to potential confusion and a lack of clarity in bounty requirements.

### Expected Behavior:
The creation process should be enhanced to ensure:
- Spaces by themselves are not accepted in bounty titles, descriptions, or acceptance criteria.
- Unnecessary spaces at the beginning and end of these fields are trimmed.
- The `Next` button is enabled only when the bounty title is validly filled, excluding fields with just spaces.

## Issue ticket number and link:
- **Ticket Number:** [ 266 ]
- **Link:** [ https://github.com/stakwork/sphinx-tribes-frontend/issues/266 ]

### Solution:
Implemented input handling improvements in the form components to ensure that leading spaces are trimmed upon input, and spaces alone are not considered valid inputs. For the acceptance criteria, the input is now processed to remove extra spaces between words and trimmed on blur to ensure the data is formatted correctly before submission.

### Changes:
`TextAreaInput.tsx (Description):` Modified onChange to use `trimStart()` for the input value.
`TextInputNew2.tsx (Title):` Adjusted to trim the start of the new value on change.
`TextAreaInputNew.tsx (Deliverables):` Enhanced onBlur and onChange handlers to trim start and normalize spaces within input values, ensuring a single space between words and trimming input both at the start and in between lines.

### Evidence:
 Please see the attached video as evidence.
 https://www.loom.com/share/7188070257264a7d8914ff55f7b01492

### Testing:
**Browser Compatibility:**
- Extensively tested on Chrome to ensure the fixes work as expected and the behavior is consistent across various test cases.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] I have tested on Chrome
- [x] I have created a unit test.
- [x] I have provided a recording

### CODE REVIEW

**Comment by elraphty** [comment]
please fix merge conflicts @aliraza556

**Comment by kevkevinpal** [comment]
@elraphty @ecurrencyhodler we should also make a similar change on sphinx-tribes to validate the input to remove any leading or trailing spaces when creating a bounty description

**Comment by ecurrencyhodler** [comment]
Good call Kev. Ali already created an issue for it here: https://github.com/stakwork/sphinx-tribes-frontend/issues/267

**Comment by kevkevinpal** [comment]
> Good call Kev. Ali already created an issue for it here: #267

we should move that issue to the backend repo

**Comment by aliraza556** [comment]
> please fix merge conflicts @aliraza556

Hi @elraphty I have resolved the conflicts. Please review now

File: src/components/form/inputs/TextAreaInputNew.tsx [hunk 1]
```diff
@@ -100,10 +100,17 @@ export default function TextAreaInputNew({
           }
           if (!textValue) {
             setShowPlaceholder(true);
+          } else {
+            const trimmedAndSingleSpacedValue = textValue
```
Comment (elraphty) [inline src/components/form/inputs/TextAreaInputNew.tsx:104]: put this in a helper function, and write a unit test for this

**Comment by elraphty** [comment]
> > please fix merge conflicts @aliraza556
> 
> Hi @elraphty I have resolved the conflicts. Please review now

Looks good, just added a few comments; move the functions to helpers.

File: src/components/form/inputs/TextInputNew2.tsx [hunk 2]
```diff
@@ -64,6 +64,10 @@ export default function TextInputNew({
     }
   }, [textValue]);
 
+  const normalizeInput = (input: string) => {
```
[the remark this answers is not in the document]
Comment (aliraza556) [reply src/components/form/inputs/TextInputNew2.tsx:67]: Alright, I'll fix it and then notify you.

File: src/components/form/inputs/TextAreaInputNew.tsx (hunk 1 above)
Comment (aliraza556) [reply src/components/form/inputs/TextAreaInputNew.tsx]: Done

File: src/components/form/inputs/TextInputNew2.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (aliraza556) [reply src/components/form/inputs/TextInputNew2.tsx]: Done

**Comment by aliraza556** [comment]
Hi @elraphty,
Please review now.

**Comment by aliraza556** [comment]
Hi @elraphty,
Please merge this PR because I will create a new one for issue #267 that addresses all aspects from both sides, meaning I will attempt to cover every edge case of the issue.

**Pull request merged**